File: Java/Intermediate/src/main/java/day46/Exercise.java
package day46;

import java.util.Arrays;
import java.util.Comparator;
import java.util.List;

public class Exercise {
    public static int ex1ElementsRemoval(int [] array) {
        int sum = 0;
        for (int j : array) {
            sum += j;
        }
        int ans = 0;
        Arrays.sort(array);
        for (int i = array.length-1; i >= 0; i--) {
            ans += sum;
            sum -= array[i];
        }
        return ans;
    }
    public static int ex2NobleInteger(int[] array) {
        Arrays.sort(array);
        int greater = 0;
        for (int i = array.length-1; i >= 0; i--) {
            if(i==array.length-1){
                greater = 0;
            }
            else if (array[i] != array[i+1]) {
                greater = array.length - 1 - i;
            }
            if (array[i] == greater) {
                return 1;
            }
        }
        return -1;
    }
    static Integer countFactors(Integer n){
        Integer count = 0;
        for(Integer i = 1 ; i * i <= n ; i++){
            if(n % i == 0){
                count++;
                if(i * i != n){
                    count++;
                }
            }
        }
        return count;
    }
    public static int[] ex3FactorsSort(int[] A) {
        List<Integer> result = Arrays.stream(A)
                .boxed().sorted(new Comparator<Integer>() {
                    @Override
                    public int compare(Integer o1, Integer o2) {
                        int count1 = countFactors(o1);
                        int count02 = countFactors(o2);
                        if(count1 == count02){
                            return Integer.compare(o1, o2);
                        }
                        return Integer.compare(count1, count02);
                    }
                }).toList();
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
    public static String ex4LargestNumber(int[] A) {
        List<Integer> result = Arrays.stream(A)
                .boxed().sorted(new Comparator<Integer>() {
                    @Override
                    public int compare(Integer o1, Integer o2) {
                        String s1 = String.valueOf(o1);
                        String s2 = String.valueOf(o2);
                        long i1=Long.parseLong(s1+s2);
                        long i2=Long.parseLong(s2+s1);
                        return Long.compare(i2, i1);
                    }
                }).toList();
        String ans = "";
        for(int i=0;i<result.size();i++){
            ans += result.get(i);
        }
        int flag = 1;
        if(ans.charAt(0) !='0'){
            return ans;
        }
        if(flag==1){
            ans =  String.valueOf(Long.parseLong(ans));
        }
        return ans;
    }
}

File: C#/src/Intermediate/day46/Exercise.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Text;

namespace Intermediate.day46
{
    public class Exercise
    {
        public static int ex1ElementsRemoval(int[] array)
        {
            array = array.OrderByDescending(x => x).ToArray();
            int sum = 0;
            for (int i = 0; i < array.Length; i++)
            {
                sum += array[i];
            }
            int ans = 0;
            for (int i = 0; i < array.Length; i++)
            {
                ans += sum;
                sum -= array[i];
            }
            return ans;
        }
        public static int ex2NobleInteger(int[] array)
        {
            array = array.OrderByDescending(x => x).ToArray();
            int greater = 0;
            for (int i = 0; i < array.Length; i++)
            {
                if (array[i] != array[i - 1])
                {
                    greater = i;
                }
                if (array[i] == greater)
                {
                    return 1;
                }
            }
            return -1;
        }
        public static int ex3Factors(int A)
        {
            int count = 0;
            for (int i = 1; i*i <= A; i++)
            {
                if(A %i == 0) { 
                    count++;
                    if(i*i != A)
                    {
                        count++;
                    }
                }
            }
            return count;
        }
        public static int[] ex3FactorsSort(int[] A)
        {
            A.Sort((x, y) =>
            {
                int c1 = ex3Factors(x);
                int c2 = ex3Factors(y);
                if (c1 == c2)
                {
                    return x.CompareTo(y);
                }
                return c1.CompareTo(c2);
            });
            return A.ToArray();
        }
        public static string ex4LargestNumber(int[] A)
        {
            A.Sort((x, y) => (y.ToString() + x.ToString()).CompareTo(x.ToString() + y.ToString()));
            string sb = A.Aggregate("", (current, t) => current + t);
            if (sb[0] == '0')
            {
                sb = double.Parse(sb).ToString(CultureInfo.CurrentCulture);
            }
            return sb;
        }
    }
}
